### PR TITLE
Add basic auth on druid JDBC driver

### LINKED ISSUE #45728: Add basic auth capabilities to Druid JDBC
Add basic auth capabilities to Druid JDBC. It is supported by Avatica driver as per [docs](https://calcite.apache.org/avatica/docs/client_reference.html#authentication). Testing requires modification to Druid's configuration mounted to the test container.

### PR DESCRIPTION
**Pull request by ddowbnac**
Closes https://github.com/metabase/metabase/issues/45728

### Description

Add basic auth capabilities to Druid JDBC.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Databases -> Add Database -> Database type -> Druid JDBC
2. Fill host data
3. Show advanced options -> Enable flag "Authentication"
4. Fill basic auth credentials

Note: A Druid database with basic authentication enabled is required to perform the full connection test. Ref: [druid-basic-security](https://druid.apache.org/docs/latest/development/extensions-core/druid-basic-security/)

### Demo

- Database properties, basic auth enabled

image

- Database properties, basic auth enabled with data

image

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj [hunk 1]
```diff
@@ -31,10 +32,13 @@
   (defmethod driver/database-supports? [:druid-jdbc feature] [_driver _feature _db] supported?))
 
 (defmethod sql-jdbc.conn/connection-details->spec :druid-jdbc
-  [_driver {:keys [host port] :as _db-details}]
+  [_driver {:keys [host port auth-enabled auth-username] :as _db-details}]
```
Comment (lbrdnk) [inline modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj:35]: It is common practice to use leading underscore in names that are not used later. As the `_db-details` are in use now, could you please remove it?

**Comment by lbrdnk** [comment]
@ddowbnac, thank you for the contribution. The change looks good! Change to docker file and image used for Druid testing would be required for meaningful automated test, hence we will do that part ourselves. Could you please sign CLA before we move forward with this PR? Instructions can be found [on "contributing" page](https://www.metabase.com/docs/latest/developers-guide/contributing).

**Comment by paoliniluis** [comment]
@ddowbnac we would need you to sign the CLA with the user that created the PR, since otherwise we won't be able to proceed due to licensing problems

**Comment by lbrdnk** [comment]
Or you could amend or rebase those commits, changing the author, to ensure all commits are made by the github user that have signed the CLA.

**Pull request closed without merging**